### PR TITLE
ZER-112 - On Admin side Calculator page add filter

### PR DESCRIPTION
**Pull request by JuliaKovtun**
dev
## JIRA

* [ZER-112](https://jira.softserve.academy/browse/ZER-112)

## Code reviewers

- [ ] @olegsobchuk 

### Second Level Review

- [ ] @github_username

## Summary of issue

As Admin I'd like to filter my list of calculators by Name or Slug

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [x]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/admins/calculators_controller.rb [hunk 1]
```diff
@@ -6,7 +6,9 @@ class CalculatorsController < BaseController
 
     def index
       @calculators = if params[:search]
-                       Calculator.where('name LIKE ?', "%#{params[:search]}%")
+                       Calculator.all.where("name ILIKE ? OR slug ILIKE ?",
```
Comment (olegsobchuk) [inline app/controllers/admins/calculators_controller.rb:9]: `if params[:search]` if here is empty string like `""` then it returns nothing, but it's wrong behavior

Comment (JuliaKovtun) [reply app/controllers/admins/calculators_controller.rb:9]: Fixed this, added `.blank?` to check if parameter is blank

Comment (olegsobchuk) [reply app/controllers/admins/calculators_controller.rb:9]: `!params[:search].blank?` => `params[:search].present?`

Comment (olegsobchuk) [reply app/controllers/admins/calculators_controller.rb:9]: 1. you don't need `.all` here
2. I don't see tests for this behavior

Comment (olegsobchuk) [reply app/controllers/admins/calculators_controller.rb:9]: 3. Remove this `if .. else` condition from controller, and use `scope` in model. Then use one `scope` here

File: app/models/calculator.rb [hunk 2]
```diff
@@ -11,4 +11,8 @@ class Calculator < ApplicationRecord
                              message: 'Only letters and numbers allowed' }
   validates :name, length: { minimum: 2 }
   validates :name, uniqueness: true
+
+  scope :by_name_and_slug, -> (params) { where("name ILIKE ? OR slug ILIKE ?",
```
Comment (olegsobchuk) [inline app/models/calculator.rb:15]: you use `OR` condition, so rename method to `by_name_or_slug`

Comment (olegsobchuk) [reply app/models/calculator.rb:15]: this approach won't work as expected. 
if `params` is `nil` or empty string it should return all records, without filtering

Comment (olegsobchuk) [reply app/models/calculator.rb:15]: rename, please, `params` to something like `name` or `search`

Comment (JuliaKovtun) [reply app/models/calculator.rb]: Renamed scope and params in last commit. If `params` is `nil` or empty string it returns all records.

**Review by olegsobchuk — APPROVED** [review]

**Review by AnastasiiaBilynska — APPROVED** [review]

**Review by anko20094 — APPROVED** [review]

**Pull request merged**